### PR TITLE
wiktionary: fix failed lookup for certain words

### PR DESCRIPTION
**Pull request by HumorBaby**
Fixes the issue sopel-irc/sopel#1361 where certain words do not return
a definition when one is available.

Words that include references in definitions expose a missed condition
check in parsing that is not relevant for most cases. References use the
`<sup>` tag, which in reference has its own `'id="'` attribute, causing
the `mode` variable in the loop to be reset, even though a definition
should be parsed from that line. These superscript references are also
now parsed out of the definition line.

### CODE REVIEW

File: sopel/modules/wiktionary.py [hunk 1]
```diff
@@ -52,7 +54,9 @@ def wikt(word):
             mode = 'particle'
         elif 'id="Preposition"' in line:
             mode = 'preposition'
-        elif 'id="' in line:
+        # 'id="' can occur in definition lines <li> when <sup> tag is used for references;
+        # make sure those are not excluded (see e.g., abecedarian).
+        elif ('id="' in line) and ('<li>' not in line):
```
Comment (dgw) [inline sopel/modules/wiktionary.py]: Could some `<li>` elements contain classes, IDs, etc. that would make this match fail? Is there any reason you've found in testing not to use `<li` without the closing `>`?

Comment (HumorBaby) [reply sopel/modules/wiktionary.py]: Good question; I was really just building off the previous logic, which, as implemented, assumes that a definition line will contain a plain `<li>` tag with no attributes or properties. However, I don't know whether this is always true  for definitions but seems to have worked for this long and I didn't want to change too much.

It is possible that `<li>` tags with classes, IDs, etc. are found elsewhere in the page, but are not part of the definition, and expanding the match to allow those may introduce more problems in the current parsing logic. 

Granted, my list of test words is very small and likely doesn't even come close to edge cases, it may be best to leave as is until another failing case is found, because this match seems to work for now (and has been). I think words are more likely to fail because of whats between the `<li>` tags (like the `<sup...>` in this case) than the `<li>` match itself.

Comment (dgw) [reply sopel/modules/wiktionary.py]: Fair enough. Let's let this sleeping dog lie. :grinning:

**Comment by dgw** [comment]
@HumorBaby Want to rebase this yourself or should I do it? It's just conflicted because I merged #1373; probably the easiest rebase ever.

**Comment by HumorBaby** [comment]
I can do it, if you don't mind; get some git practice in.... rebase to follow in a few mins.

**Comment by HumorBaby** [comment]
Ah snap, forgot to change the commit message! force-push coming right up

**Comment by dgw** [comment]
> I can do it, if you don't mind

Actually I prefer when PR authors do their own rebases, because it avoids me getting my info plastered onto commits that I didn't create. Git still maintains the "Author" details as-is, but I wind up being the "Committer" and my avatar/name gets displayed in the history when I rebase other people's work.

Sopel just seems to have a lot of contributors who only use GitHub's web UI, which doesn't support rebasing, so I've gotten used to asking. :smile_cat:

**Comment by HumorBaby** [comment]
New conflict from #1405 merge. Rebasing to resolve...

File: sopel/modules/wiktionary.py [hunk 2]
```diff
@@ -13,13 +13,15 @@
 from sopel import web
 from sopel.module import commands, example
 
-uri = 'https://en.wiktionary.org/w/index.php?title=%s&printable=yes'
+uri = 'http://en.wiktionary.org/w/index.php?title=%s&printable=yes'
```
Comment (dgw) [inline sopel/modules/wiktionary.py]: This reverts #1373… Can we not do that? :smile_cat:

Comment (HumorBaby) [reply sopel/modules/wiktionary.py]: Gotta remember to check my diffs for unintended changes :confused:
